sqlx-core/src/mysql/connection.rs

```diff
@@ -181,7 +181,11 @@ async fn establish(stream: &mut MySqlStream, url: &Url) -> crate::Result<()> {
             HandshakeResponse {
                 client_collation: COLLATE_UTF8MB4_UNICODE_CI,
                 max_packet_size: MAX_PACKET_SIZE,
-                username: url.username().unwrap_or("root"),
+                username: url
+                    .username()
+                    .map(|c| c.into_owned())
+                    .unwrap_or(String::from("root"))
```

Comment:
You want to say `unwrap_or(Cow::Borrowed("root")).as_str()` to avoid the allocations here. 

As an aside, with #174 we can centralize a lot of this tweaking.

Reply:
Good shout. I've applied this change. I had a quick go at doing it for `postgres/connection.rs` but didn't get very far. I can fight with it a bit more tomorrow or leave it as is if you're ok with that.

Reply:
No worries. I can merge as-is and take a minute to see how possible it is. It should get refactored with #174 anyway.